OOP/OOP6_ResizableArray/src/main/java/LukCharList.java

```diff
@@ -0,0 +1,146 @@
+public class LukCharList implements CharList {
+
+    public char[] array = new char[0];
+    public char[] temporaryArray = new char[0];
+
+    public LukCharList(){
+    }
+
+    public int save(char charToSave) {
+        if(this.array.length > 0) {
+            temporaryArray = this.array;
+        }
+        this.array = new char[this.array.length + 1];
+        rewrite(temporaryArray);
+        array[this.array.length - 1] = charToSave;
+        return this.array.length - 1;
+    }
+
+    private char[] rewrite(char[] temporary) {
+        if(temporary.length > 0) {
+            for(int newLength = 0; newLength < temporary.length; newLength++) {
+                this.array[newLength] = temporary[newLength];
+            }
+        }
+        return this.array;
```

Comment:
why return this.array? What's the purpose of this return, since it's a class field?

Reply:
@MatDls I remove this, it's redundant